include/keisan/number.impl.hpp

```diff
@@ -53,7 +53,11 @@ T map(
   const T & value, const T & source_min, const T & source_max,
   const T & target_min, const T & target_max)
 {
-  return target_min + scale(value - source_min, source_max - source_min, target_max - target_min);
+  auto source_val = value;
+  source_val = std::min(source_val, std::max(source_min, source_max));
+  source_val = std::max(source_val, std::min(source_min, source_max));
+  
+  return target_min + scale(source_val - source_min, source_max - source_min, target_max - target_min);
```

Comment:
The `map` function is supposed to allow the source or target values to be outside the min/max bounds, similar to the `scale` function.

To keep the values inside the min/max bounds, I suggest manually clamping the values prior to calling the `map` function instead of always forcing the `map` function to clamp the values.

You may also need to check other projects that utilize this function to see if forcing clamping in the `map` function is safe and does not affect the original logic of those projects.

Reply:
CMIIW. I have different idea about the logic of the ```map``` function. The purpose of this ```map``` function is to return value between ```target_min``` and ```target_max``` based on where the ```source``` value is located between ```source_min``` and ```source_max```. By enforcing clamping, we ensure that the return values is between  ```target_min``` and ```target_max```. Without enforcing clamping, the returned value could be outside the target range, which I don't think is the intended behavior of the function.

Additionally, enforcing clamping in the ```map``` function is what has been implemented in the old framework in [algebra.cpp](https://github.com/ichiro-its/framework/blob/master/library/common/src/algebra.cpp)

Reply:
Got it. Apparently, it's been like this all these years in the old framework. Just make sure that this change doesn't cause side effects on other packages that use this function.